cmd/stats-exporter/main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+type sqlRows interface {
+	Next() bool
+	Scan(dest ...interface{}) error
+}
+
+type dbQueryable interface {
+	Query(string, ...interface{}) (*sql.Rows, error)
+}
+
+var sqlOpen = func(driver, dsn string) (dbQueryable, error) {
+	return sql.Open(driver, dsn)
+}
+
+var execRun = func(c *exec.Cmd) error {
+	return c.Run()
+}
+
+func queryDB(dbConnect, beginTimeStamp, endTimeStamp string) (*sql.Rows, error) {
+	dbDSN, err := ioutil.ReadFile(dbConnect)
+	if err != nil {
+		return nil, fmt.Errorf("Could not open database connection file: %s", err)
+	}
+	db, err := sqlOpen("mysql", strings.TrimSpace(string(dbDSN)))
+	if err != nil {
+		return nil, fmt.Errorf("Could not establish database connection: %s", err)
+	}
+	rows, err := db.Query(`SELECT id,reversedName,notBefore,serial FROM issuedNames where notBefore >= ? and notBefore < ?`, beginTimeStamp, endTimeStamp)
+	if err != nil {
+		return nil, fmt.Errorf("Could not reach database and run query: %s", err)
+	}
+	return rows, nil
+}
+
+func writeTSVData(rows sqlRows, outFile io.Writer) error {
+	for rows.Next() {
```

Comment:
Should `writeTSVData` do something differently if `rows.Next()` returns false on the first call? e.g. that there are no result rows. Maybe return an error?

Reply:
@jsha what are your thoughts here? I am indifferent, but you will be processing the data. Will an empty file be a problem? Is it preferable to error out instead?

Reply:
I think it makes sense to error out rather than produce empty results; that's more likely to get our attention if something is wrong.